ASP_Threads - implement webApi back-end and WPF front-end applications

Trambler-2 opened the pull request.

Trambler-2 committed:
ASP_Threads - implement webApi back-end and WPF front-end applications

troepolik commented on ASP_Threads/Back-end/WebApiTest/Extensions/TaskActionExtensions.cs:
```diff
@@ -0,0 +1,47 @@
+                .ContinueWith((t, o) => actionIfCancel(), null, TaskContinuationOptions.OnlyOnCanceled);
+        public static async Task After(this IEnumerable<Task> tasks, Action afterWork)
+        {
+            var actions = tasks.ToList();
```
зачем тут к листу приведение?

troepolik commented on ASP_Threads/Back-end/WebApiTest/Extensions/TaskActionExtensions.cs:
```diff
@@ -0,0 +1,47 @@
+            while (counter++ != iterations)
+            {
+                var task1 = Task.Factory.StartNew(action);
+                var task2 = Task.Factory.StartNew(withAction);
```
стартовать на каждой итерации новый поток нецелесообразно.
у тебя каждая таска должна свои итерации делать и они у неё могут быть совершенно свои.
синхронизация должна быть внутри каждой таски

troepolik commented on ASP_Threads/Back-end/WebApiTest/Extensions/TaskActionExtensions.cs:
```diff
@@ -0,0 +1,47 @@
+{
+    public static class TaskActionExtensions
+    {
+        public static Task RunWithCancellation(this Action<CancellationToken> action, CancellationToken token, Action actionIfCancel) =>
```
вижу какие-то экстеншены, а как исопльзовались не вижу.
не вижу где ты написал задачу, которая прекращает свое действие по команде

Trambler-2 committed:
ASP_Threads - task 2

replaced default logging to Log4net, added task/action extensions, added db Provider.
ASP_Threads - task 2 updated

changed priority of providers
ASP_Threads_tasks 2

Trambler-2 committed:
ASP_Threads - update db fetch

troepolik commented on ASP_Threads/Back-end/WebApiTest/Startup.cs:
```diff
@@ -18,11 +19,16 @@ public class Startup
+                .AddDbSettings<SqlConnection>(configuration.GetConnectionString("ConnectionString"),
+                    "select * from Settings")
+                .AddConfiguration(configuration)
+                .Build();
```
к сожалению в Startup так делать нельзя.
эта новая конфигурация будет доступна только в стартапе.
если я вытащу из DI  IConfiguration , то там будет старый конфигурейшен.
сделать это нужно именно в Program

Trambler-2 committed:
ASP_Threads task 2 remove db from Startup
